From dcc452a6a41c6be294d45a7da467a611851bc879 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 24 Apr 2013 09:35:00 +0300
Subject: [PORT FROM MAIN] Telephony: Retrieve IMEI, IMEISV, baseband on "radio on"

BZ: 84990

getIMEI, getIMEISV and getBasebandVersion are
not retrieved on flight mode.

As per framework RADIO_OFF is also considered
as radio available. Due to this, getIMEI, getIMEISV
and getBasebandVersion are getting triggered on
radio off. So, this requests are completed with
error.

With this patch, IMEI, IMEISV and Baseband version
is retrieved on "radio on" event.

Change-Id: I932f4a22c244639a4570a3c17e2e6a308f1d52d0
Orig-Change-Id: I008c624079d9d95ffaf5b5bfd2b801e9dc260ba3
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../android/internal/telephony/gsm/GSMPhone.java   |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GSMPhone.java b/src/java/com/android/internal/telephony/gsm/GSMPhone.java
index 9a4021b..2814f72 100644
--- a/src/java/com/android/internal/telephony/gsm/GSMPhone.java
+++ b/src/java/com/android/internal/telephony/gsm/GSMPhone.java
@@ -1216,6 +1216,7 @@ public class GSMPhone extends PhoneBase {
         Message onComplete;
 
         switch (msg.what) {
+            case EVENT_RADIO_ON:
             case EVENT_RADIO_AVAILABLE: {
                 mCi.getBasebandVersion(
                         obtainMessage(EVENT_GET_BASEBAND_VERSION_DONE));
@@ -1225,9 +1226,6 @@ public class GSMPhone extends PhoneBase {
             }
             break;
 
-            case EVENT_RADIO_ON:
-            break;
-
             case EVENT_REGISTERED_TO_NETWORK:
                 syncClirSetting();
                 break;
-- 
1.7.4.1

